src/Servers/Kestrel/test/InMemory.FunctionalTests/Http2/Http2StreamTests.cs

```diff
@@ -103,22 +103,41 @@ public async Task HEADERS_Received_KnownOrCustomMethods_Accepted(string method)
             };
         await InitializeConnectionAsync(_echoMethodNoBody);
 
+        // First request
         await StartStreamAsync(1, headers, endStream: true);
 
-        var headersFrame = await ExpectAsync(Http2FrameType.HEADERS,
+        var headersFrame1 = await ExpectAsync(Http2FrameType.HEADERS,
             withLength: 45 + method.Length,
             withFlags: (byte)(Http2HeadersFrameFlags.END_HEADERS | Http2HeadersFrameFlags.END_STREAM),
             withStreamId: 1);
 
-        await StopConnectionAsync(expectedLastStreamId: 1, ignoreNonGoAwayFrames: false);
+        _hpackDecoder.Decode(headersFrame1.PayloadSequence, endHeaders: false, handler: this);
 
-        _hpackDecoder.Decode(headersFrame.PayloadSequence, endHeaders: false, handler: this);
+        Assert.Equal(4, _decodedHeaders.Count);
+        Assert.Contains("date", _decodedHeaders.Keys, StringComparer.OrdinalIgnoreCase);
+        Assert.Equal("200", _decodedHeaders[HeaderNames.Status]);
+        Assert.Equal("0", _decodedHeaders[HeaderNames.ContentLength]);
+        Assert.Equal(method, _decodedHeaders["Method"]);
+        _decodedHeaders.Clear();
+
+        // Second request (will use dynamic table indexes)
+        await StartStreamAsync(3, headers, endStream: true);
+
+        var headersFrame2 = await ExpectAsync(Http2FrameType.HEADERS,
+            withLength: 7,
```

Comment:
The test will fail when run in release now right?

Reply:
I'm pretty sure our PR checks run in release, so apparently not. I guess the DEBUG-only changes aren't strictly necessary for the test to pass. Can you explain this a little more @JamesNK?

Reply:
Response headers are always encoded with static compression. The hack in this or is to get request headers - :method in this case - to use static compression